src/parser/jobs/blm/Gauge.js

```diff
@@ -0,0 +1,285 @@
+					this._eno = 0
+					this._enoTimer = 0
+					this._droppedEno ++
+				}
+			}
+			else {
+				this._AFUITimer = event.timestamp
+				this.AF ++
+				this.AF = Math.min(this.AF,3)
+			}
+		}
+
+		//check for UI1 actions and update buffs accordingly. Also check for lost fouls and dropped enos while you're at it.
+		if (UI1_ACTIONS.includes(abilityId)) {
+			if (this._AF > 0) {
+				this._AF = 0
+				this._UI = 0
+				this._UH = 0
+				this._AFUITimer = 0
+				if (this._poly === 0 && this._eno > 0) {
+					this._eno = 0
+					this._enoTimer = 0
+					this._droppedEno ++
+					this._lostFoul ++
+				}
+				if (this._poly > 0 && this._eno > 0) {
+					this._eno = 0
+					this._enoTimer = 0
+					this._droppedEno ++
+				}
+			}
+			else {
+				this._AFUITimer = event.timestamp
+				this._UI ++
+				this._UI = Math.min(this._UI,3)
+			}
+		}
+			
+		//do F3 things
+		if (abilityId === ACTIONS.FIRE_III.id) {
+			this._UI = 0
+			this._AF = 3
+			this._AFUITimer = event.timestamp
+		}
+
+		//do B3 things
+		if (abilityId === ACTIONS.BLIZZARD_III.id) {
+			this._UI = 3
+			this._AF = 0
+			this._AFUITimer = event.timestamp
+		}
+
+		//Foul support and poly adjustment
+		if (abilityId === ACTIONS.FOUL.id) {
+			this._poly = 0
+
+		}
+
+		if(abilityId === ACTIONS.TRANSPOSE.id) {
+			if (this._AF > 0) {
+				this._AF = 0
+				this._UI = 1
+				this._AFUITimer = event.timestamp
+			}
+			if (this._UI > 0) {
+				this._UI = 0
+				this._AF = 1
+				this._AFUITimer = event.timestamp
+			}
+
+		}
+
+		//keep track of UH
+		if (abilityId === ACTIONS.BLIZZARD_IV.id) {
+			this._UH = 3
+		}
+		//getting rid of UHs one AF action at a time
+		if (AF_ACTIONS.includes(abilityId) && this._UH > 0) {
+			this._UH --
+			this._UH = Math.max(this._UH, 0)
+		}
+
+		//Flare resetting UHs
+		if (abilityId === ACTIONS.FLARE.id && this._UH > 0) {
+			this._UH = 0
+		}
+
+		
+	}
+	_onDeath() {
+		// Death just flat out resets everything except for poly. Rip.
+		// Not counting the loss towards the rest of the gauge loss, that'll just double up on the suggestions
+		this._AF = 0
+		this._UI = 0
+		this._UH = 0
+		this._AFUITimer = 0
+		this._eno = 0
+		this._enoTimer = 0
+		}
+
+	_onComplete() {
+		// Suggestions for lost eno
+		if (this._droppedEno) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.ENOCHIAN.icon,
+				content: <Fragment>
+					Dropping <ActionLink {...ACTIONS.ENOCHIAN}/> may lead to lost <ActionLink {...ACTIONS.FOUL}/>, more clipping because of additional <ActionLink {...ACTIONS.ENOCHIAN}/> casts, unavailability of <ActionLink {...ACTIONS.FIRE_IV}/> and <ActionLink {...ACTIONS.BLIZZARD_IV}/> or straight up missing out on the 10% damage bonus that <ActionLink {...ACTIONS.ENOCHIAN}/> provides.				
+				</Fragment>,
+				severity: SEVERITY.MAJOR,
+				why: <Fragment>
+					You dropped Enochian {this._droppedEno} time{this._droppedEno > 1 && 's'}.
+				</Fragment>,
+			}))
+		}
+
+		if (this._lostFoul) {
+			this.suggestions.add(new Suggestion({
```

Comment:
Lost and Overwritten foul suggestions are quite similar. Would it be worth combining the two and just showing both data points in the `why`?

Reply:
this will stay as it is since the outcome is the same (you lost fouls) but the reason why is important since it addresses different problems in a BLMs gameplay. I am aware that they are worded quite simliar